promql/engine.go

```diff
@@ -514,15 +493,21 @@ func (ng *Engine) populateIterators(ctx context.Context, s *EvalStmt) (storage.Q
 
 	mint := s.Start.Add(-maxOffset)
 
-	querier, err := ng.queryable.Querier(ctx, timestamp.FromTime(mint), timestamp.FromTime(s.End))
+	querier, err := q.Querier(ctx, timestamp.FromTime(mint), timestamp.FromTime(s.End))
 	if err != nil {
 		return nil, err
 	}
 
-	Inspect(s.Expr, func(node Node) bool {
+	Inspect(s.Expr, func(node Node, path []Node) bool {
+		params := &storage.SelectParams{
+			Step: int64(s.Interval / time.Millisecond),
+		}
+
 		switch n := node.(type) {
 		case *VectorSelector:
```

Comment:
Can we reuse some code here? The code for Vector and Matrix is totally the same except:
`durationMilliseconds(LookbackDelta)` vs `durationMilliseconds(n.Range)`

Reply:
I recall sitting over this quite a bit back than and all re-usages turned out complicated and/or ugly.
Mainly because we get the type assurance through the type-switch statement and extracting something requires us to create another helper interface just for this and a function somewhere else.

The reuslting cognitive overhead just seemed worse.